hid-ghlive/src/hid-ghlive.c

```diff
@@ -41,121 +41,28 @@ static const char ghl_ps4_magic_data[] = {
 struct ghlive_sc {
 	struct hid_device *hdev;
 	unsigned long quirks;
-	int device_id;
+	int device_id; /* What is this for? */
+	struct urb *urb;
 	struct timer_list poke_timer;
-	struct usb_ctrlrequest *cr;
-	struct usb_endpoint_descriptor *ep_irq_out;
-	u8 *databuf;
 };
 
-static void ghl_magic_poke_cb_ps3wiiu(struct urb *urb)
+static void ghl_magic_poke_cb(struct urb *urb)
 {
-	if (urb) {
-		/* Free sc->cr and sc->databuf allocated in ghl_magic_poke_ps3wiiu() */
-		kfree(urb->setup_packet);
-		kfree(urb->transfer_buffer);
-	}
-}
+	struct ghlive_sc *sc = urb->context;
```

Comment:
Pretty sure there should first be a `if (urb)`. In case that fails, I guess we should throw a `hid_err()`.

Reply:
I will do : 
``` c
static void ghl_magic_poke_cb(struct urb *urb)
{
	struct ghlive_sc *sc = urb->context;

	if (urb->status < 0)
		hid_err(sc->hdev, "URB transfer failed : %d", urb->status);
		
	mod_timer(&sc->poke_timer, jiffies + GHL_GUITAR_POKE_INTERVAL*HZ);
}
```

Reply:
So if it fails we retry. Sounds right.